### ISSUE TITLE
Unused Variable Analysis Does Not Account for Local Function Default Parameter Values

### ISSUE THREAD

**Issue by adamperlin** [dotnet/roslyn#63837]
Prompted by the following example:

```csharp
using System;

public class C {
    const int Const = 10;

    public static void M() {
        const int n = 3;
        static int Fun1(int x = n) => x;
        static int Fun2(int x = Const) => x;
        
        Console.WriteLine(Fun1());
        Console.WriteLine(Fun2());
    }
}
```

(A [sharplab](https://sharplab.io/) repro is ideal for compiler/language issues whenever possible)
https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzguATEBqAPgAQCYBGAWACh8BmAAiJoGEaBvCm9mgYwHsY0bYCRr34BeGsQAMAbjYdqdYgDY6AFhoBZABQBKFnI4cefIYJowa4qrPKHD+ZQJhCAYgFcYxLWYAel83qiAHw0PjZ28o5m7jCE3s6h/gwiCIEhYQZ2mfbEAJxaMV46OuERivkxccWl7AC+FLVAA===

**Expected Behavior**:
The local constant `n` should be shown as "used" since it is the default value for a local function parameter. The class constant `Const` should also be shown as used for the same reason. 

**Actual Behavior**:
`n` and `Const` are shown as unused, and the diagnostic `warning CS0219` for an unused variable is raised.

**Comment by adamperlin** [comment]
Ideally, any fix for the analysis here should also allow constant usages to be tracked in lambda default parameter values
